include/spdk/nvme.h

```diff
@@ -374,6 +379,26 @@ void spdk_nvme_ctrlr_register_aer_callback(struct spdk_nvme_ctrlr *ctrlr,
 struct spdk_nvme_qpair;
 
 /**
+ * Signature for the callback function invoked when a timeout is
+ * detected on a request.
+ */
+typedef void (*spdk_nvme_timeout_cb)(struct spdk_nvme_ctrlr *ctrlr,
+				     struct spdk_nvme_qpair *qpair);
+
+/**
+ * \brief Register for timeout callback on a controller.
+ *
+ * The application can choose to register for timeout callback or not register
+ * for timeout callback.
+ *
+ * \param ctrlr NVMe controller on which to monitor for timeout.
+ * \param cb_fn A function pointer that points to the callback function
+ * \param cb_arg Argument to the callback function.
```

Comment:
The documentation mentions cb_arg, but there is no such parameter to this function.  I think we should add one to match the other NVMe library callbacks and to allow the caller to pass through an opaque context pointer.

Also, documentation for the timeout field is missing.  What units? How does this timeout value interact with the one specified in spdk_nvme_ctrlr_opts?